tests/state/test_state_transition.py

```diff
@@ -173,3 +189,94 @@ def test_recent_proposer_processing(genesis_crystallized_state):
     assert some_deltas[0] == 100
     assert some_deltas[5] == 200
     assert some_deltas[39] == 300
+
+
+@pytest.mark.parametrize("balances, votes, justifying", [
+    ([1, 1, 1], [True, True, True], True),
+    ([1, 1, 1], [True, True, False], True),
+    ([1, 1, 1], [True, False, True], True),
+    ([1, 1, 1], [False, True, True], True),
+    ([1, 1, 1], [True, False, False], False),
+    ([1, 1, 1], [False, True, False], False),
+    ([1, 1, 1], [False, False, True], False),
+    ([1, 1, 1], [False, False, False], False),
+
```

Comment:
I'd like to see at least a couple of tests for a validator set size greater than 3. 4 and/or 5 should work :)